src/apiServices/AnalysesService.ts

```diff
@@ -0,0 +1,106 @@
+import { postRequest, getRequest } from '../http'
+
+import { errorHandler } from '../util/errorHandler'
+import { getHeaders } from '../util/getHeaders'
+import { submitBytecodeRequest } from '../util/submitBytecodeRequest'
+
+import { getTokensNode } from '../node'
+
+import { API_URL_PRODUCTION, API_URL_STAGING, tokenLocation } from '../util/constants'
+
+import { JwtTokensInterface, SubmitContractRes } from '..'
+
+import { submitContractRequest } from '../analyses/submitContractRequest'
+
+export class AnalysesService {
+    public token
+    private apiUrl: string = API_URL_PRODUCTION
+
+    constructor(token?) {
+        this.token = token
+    }
+
+    public async getAnalysesList() {
+        try {
+            let headers;
+            if (this.token) {
+                headers = getHeaders(this.token)
+            } else {
+                const jwtTokens: JwtTokensInterface = getTokensNode(tokenLocation)
+                headers = getHeaders(jwtTokens.access)
+            }
+
+            const result = await getRequest(`${this.apiUrl}/analyses`, headers)
+            console.log('getAnalysesList result', result.data)
+            return result.data
+        }
+        catch (err) {
+            errorHandler(err)
+        }
+    }
+
+    public async getAnalysisStatus(uuid: string) {
+        try {
+            let headers;
+            if (this.token) {
+                headers = getHeaders(this.token)
+            } else {
+                const jwtTokens: JwtTokensInterface = getTokensNode(tokenLocation)
+                headers = getHeaders(jwtTokens.access)
+            }
+
+            const result = await getRequest(`${this.apiUrl}/analyses/${uuid}`, headers)
```

Comment:
We can probably write a wrapper for the `getRequest` and `postRequest` methods that read the this.token or this.headers and sends the request with the correct headers, as we do reuse that logic in multiple places.